packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1494,15 +1498,21 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
      * @param fullTree - true to bypass optimizations and force a full summary tree.
      * @param trackState - This tells whether we should track state from this summary.
      */
-    private async summarize(fullTree: boolean = false, trackState: boolean = true): Promise<ISummaryTreeWithStats> {
+    private async summarize(fullTree: boolean = false, trackState: boolean = true): Promise<IChannelSummarizeResult> {
         // Summarizer node tracks the state from the summary. If trackState is true, use summarizer node to get
         // the summary. Else, get the summary tree directly.
-        return trackState
+        const summarizeResult = trackState
             ? await this.summarizerNode.summarize(fullTree) as ISummaryTreeWithStats
             : await this.summarizeInternal(fullTree, false /* trackState */) as ISummaryTreeWithStats;
+        return {
+            stats: summarizeResult.stats,
+            summary: summarizeResult.summary,
+            nodes: this.garbageCollectionNodes,
```

Comment:
I'm a bit at a loss on how to read this code.
The fact that GC info comes from the side, not as a result of running a summary suggests that it's out of sync with summary (or can be) and that we potentially have wrong data to work with.
Can we instead plumb through proper interface to directly return that data and not rely on ambient state?

Reply:
This will need some additional work to make it right. I will do this in a separate PR.